psyneulink/core/compositions/composition.py

```diff
@@ -10477,11 +10477,11 @@ def _execute_controller(self,
                        context=context,
                        node=self.controller)
 
-                if self.controller and not execution_mode:
+                if self.controller and not execution_mode & pnlvm.ExecutionMode.COMPILED:
                     context.execution_phase = ContextFlags.PROCESSING
                     self.controller.execute(context=context)
 
-                if execution_mode:
+                if execution_mode & pnlvm.ExecutionMode.COMPILED:
```

Comment:
This (and all other checks before `execute_node`)  can use an assertion:
`assert execution_mode == pnlvm.ExecutionMode.LLVM or execution_mode & pnl.ExecutionMode._Fallback`
Those two cases are the only way to reach the code.

Reply:
Applied assertion here, but other checks (at least on composition.py) seem to be part of legitimate branching conditions, not simple else's.